src/gsNurbs/gsKnotVector.hpp

```diff
@@ -65,12 +68,21 @@ class gsXml< gsKnotVector<T> >
             }
         }
 
-        // Case: mode: none/default
-        std::istringstream str;
-        str.str( node->value() );
-        for (T knot; gsGetReal(str, knot);)
-            knotValues.push_back(knot);
-
+        // Read knots from list
+        gsXmlAttribute* format = node->first_attribute("format");
+        std::string format_flag = format ? format->value() : "ascii";
+        std::transform(format_flag.cbegin(), format_flag.cend(),
+                       format_flag.begin(),
+                       [](unsigned char c) { return std::tolower(c); });
```

Comment:
Please use "::tolower" instead of "std::tolower" because some windows compiles like it better.